service-block/src/main/java/blocks/service/AbstractBlock.java

```diff
@@ -13,6 +13,11 @@
         });
     }
 
+    @Override
+    public BlockRef<T> ref() {
+        return getRef(this.getClass().getName());
```

Comment:
Just wondering if it would make sense to have a single instance of reference but it doesn't keep any state so unless someone calls this excessively it should be fine...